narwhal/primary/src/synchronizer.rs

```diff
@@ -251,7 +252,18 @@ impl Synchronizer {
         certificate: Certificate,
         network: &Network,
     ) -> DagResult<()> {
-        self.process_certificate_internal(certificate, network)
+        self.process_certificate_internal(certificate, network, true)
+            .await
+    }
+
+    /// Tries to accept a certificate that is already sanitized.
+    // TODO: produce a different type after sanitize, e.g. VerifiedCertificate.
```

Comment:
I like the idea of using a new type we can check against instead of passing a bool around